### ISSUE TITLE
number of arguments expected by dma_buf_export function

### ISSUE THREAD

**Issue by seungeun2** [OP-TEE/optee_linuxdriver#26]
I ran "setup_juno_optee.sh" script(got it from optee_os git) and encountered an error when building the linux driver.

The error was about dma_buf_export function called in "core/tee_shm.c".
It complains about the number of arguments passed to the function (5 arguments are passed when it expects 4.)

          In core/tee_shm.c
          dmabuf = dma_buf_export(shm, &_tee_shm_dma_buf_ops, shm->size_alloc, O_RDWR, 0);

Indeed according to the linux source code(linux/include/linux/dma-buf.h) dma_buf_export macro function expects just 4 arguments.

         #define dma_buf_export(priv, ops, size, flags) \
                dma_buf_export_named(priv, ops, size, flags, KBUILD_MODNAME)

I thought that the 5th argument might be unnecessary in that its value is 0, so I removed it from the linux driver code.
After this modification I could build the linux driver just fine and it seems to work properly when actually running CA and TA.

My question here is whether the modification I made is valid.

It seems that the interface(type signature) of "dma_buf_export" function differs by linux versions.

To make the driver more compatible, how about put some "if def"  and determine the version of "dma_buf_export" function to be called at compile time?
(-> maybe it's not possible since the driver is built using liux kernel build system? Just suggesting!)

**Comment by MrVan** [comment]
You may remove the last one. 
       dmabuf = dma_buf_export(shm, &_tee_shm_dma_buf_ops, shm->size_alloc,
-                               O_RDWR, 0);
+                               O_RDWR);

**Comment by MrVan** [comment]
The Travis CI build failed, will anyone fix this?
```
CC [M]  /home/travis/build/OP-TEE/optee_linuxdriver/core/tee_mutex_wait.o
/home/travis/build/OP-TEE/optee_linuxdriver/core/tee_shm.c: In function ‘export_buf’:
/home/travis/build/OP-TEE/optee_linuxdriver/core/tee_shm.c:365:14: error: macro "dma_buf_export" passed 5 arguments, but takes just 4
     O_RDWR, 0);
              ^
/home/travis/build/OP-TEE/optee_linuxdriver/core/tee_shm.c:364:11: error: ‘dma_buf_export’ undeclared (first use in this function)
  dmabuf = dma_buf_export(shm, &_tee_shm_dma_buf_ops, shm->size_alloc,
           ^
/home/travis/build/OP-TEE/optee_linuxdriver/core/tee_shm.c:364:11: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [/home/travis/build/OP-TEE/optee_linuxdriver/core/tee_shm.o] Error 1
make[2]: *** Waiting for unfinished jobs....
  LD [M]  /home/travis/build/OP-TEE/optee_linuxdriver/armtz/optee_armtz.o
make[1]: *** [/home/travis/build/OP-TEE/optee_linuxdriver/core] Error 2
make: *** [_module_/home/travis/build/OP-TEE/optee_linuxdriver] Error 2
```

**Comment by jbech-linaro** [comment]
Hi @seungeun2  and @MrVan we are aware of this issue and the fix is easy. If you are using older kernels then you don't have to make any changes. If you are using newer kernels (4.x something), then you need to make the changes that could be found in #21 request.

The reason for us not having merged this yet is because we're kind of stuck between different kernel versions for various setups. Ideally we would like to step up to us the latest kernel, but since we are dependent on where ARM-TF are, we cannot just stick to latest. Until this has been officially fixed. I recommend that you are using the fix in #21 if you are using later kernels.

Sorry for the (temporary) inconvenience caused by this.

// Joakim
